### DISCUSSION TITLE
How to get grenade throw position?

### DISCUSSION THREAD

**Discussion by tugamars** [saul/demofile#246] [Questions] [answered]
Hi, first of all thanks for this fantastic library.

I have found a problem ( i think ), the event called "grenade_thrown" is never called, so there is no way of catching the position the nade/flash/smoke/etc was thrown from, right?

Is there any other event that can be used for that effect?

**Comment by clappis** [comment]
I have the same problem, only works this events:

hegrenade_detonate, flashbang_detonate and smokegrenade_detonate

**Comment by saul** [comment, marked as answer]
Try some code like this:

```ts
demoFile.entities.on("create", e => {
  if (!("DT_BaseCSGrenadeProjectile" in e.entity.props)) return;

  const projectileEntity = (e.entity as unknown) as BaseEntity<
    CBaseCSGrenadeProjectile
  >;
  const thrower = projectileEntity.owner as Player;

  console.log(
    "%s threw %s at:",
    thrower ? thrower.name : "(someone)",
    projectileEntity.modelName,
    projectileEntity.position
  );
});
```

**Comment by EloB** [comment]
@saul Is it possible to get `Weapon` instead of `BaseEntity`?

**Comment by saul** [comment]
Grenade projectiles aren't weapons, so unfortunately not. That is projectiles (e.g. `DT_DecoyProjectile`) do not inherit from `DT_WeaponCSBase`
